applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp

```diff
@@ -296,8 +296,8 @@ namespace Kratos
 
             // We set the database
             ModelPart& r_computing_contact_model_part = rModelPart.GetSubModelPart("ComputingContact");
-            for (auto p_slave_cond : slave_conds) {
-                for (auto p_master_cond : master_conds) {
+            for (const auto& p_slave_cond : slave_conds) {
+                for (const auto& p_master_cond : master_conds) {
                     id_cond++;
                     const PairedCondition& r_reference_condition = dynamic_cast<const PairedCondition&>(KratosComponents<Condition>::Get("ALMFrictionalMortarContactCondition3D4N"));
```

Comment:
beware, this Get is not fast. You should rather get it once and then clone it as you need

Reply:
(also i don't really understand what you are doing here)